src/OpenTracing/LogUtils.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace OpenTracing\LogUtils;
+
+use Exception;
+use InvalidArgumentException;
+use OpenTracing\LogField;
+use Throwable;
+
+function interleavedKVToFieldsConverter(array $keyValueFields)
+{
+    $fields = [];
+
+    foreach ($keyValueFields as $key => $value) {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(sprintf("Non-string key, got %s of type %s", $key, gettype($key)));
+        }
+
+        switch (true) {
+            case is_bool($value):
+                $fields[] = LogField::asBool($key, $value);
+                break;
+            case is_float($value) || is_numeric($value):
+                $fields[] = LogField::asFloat($key, $value);
+                break;
+            case is_int($value):
+                $fields[] = LogField::asInt($key, $value);
+                break;
+            case is_string($value):
+                $fields[] = LogField::asString($key, $value);
+                break;
+            case ($value instanceof Exception):
```

Comment:
Not in versions prior 7.0. So in this version I use a polifyll for `Throwable` but that is only the interface declaration.
In PHP 5.6 + polifyll, exception does not implement Throwable.